src/render.rs

```diff
@@ -185,6 +203,22 @@ pub fn render_encoding_full(
     let scene_buf = ResourceProxy::Buf(recording.upload("scene", packed.data));
     let config_buf =
         ResourceProxy::Buf(recording.upload_uniform("config", bytemuck::bytes_of(&config)));
+    let info_size = config.layout.bin_data_start;
+    let bump = BumpAllocators {
+        binning_size: ((1 << 20) / 4) - info_size,
+        ptcl_size: (1 << 25) / 4,
```

Comment:
I'd be tempted to do `size_of::<u32>()` here and above, just to make things really obvious

My initial reaction when seeing this was "why not just not shift left as far". I see that this version shows in-source the exact sizes of the buffers, whereas the version which doesn't shift as far then hides the buffer sizes.

Reply:
Yes, the size calculation code is nasty across the board. We will eventually replace this with better heuristics for initial sizes and typed buffer handles that handle element sizing automatically.